src/network.rs

```diff
@@ -632,3 +632,22 @@ mod transport {
         Ok(transport)
     }
 }
+
+#[cfg(test)]
+mod arbitrary {
+    use super::*;
+    use libp2p::multihash;
+    use quickcheck::{Arbitrary, Gen};
+
+    impl Arbitrary for Taker {
```

Comment:
I can recommend looking into proptest as an alternative to quickcheck: https://docs.rs/proptest/0.10.0/proptest/

To nice thing is, it doesn't really on traits being implemented, you are just composing functions together.

Reply:
proptest does not support async test, it's why I opted for quickcheck.

Reply:
ah bummer :/